### PR TITLE
5634 new unit tests

### PR DESCRIPTION
**Pull request by nikzaugg**
This pull request contributes three new test classes with unit tests (as mentioned in #5634 ).

## Related Issues

- connects to #5634 

## Pull Request Checklist

- [x] Unit [tests][] completed
- [x] Integration [tests][]: None
- [x] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [x] [Documentation][docs] completed
- [x] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: src/test/java/edu/harvard/iq/dataverse/mydata/MyDataUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+@RunWith(Theories.class)
+public class MyDataUtilTest {
```
Comment (matthew-a-dunlap) [inline src/test/java/edu/harvard/iq/dataverse/mydata/MyDataUtilTest.java]: I hadn't seen JUnit theories before. They look really powerful and I like how you've used them here. They are experimental though and it looks like JUnit 5 is steering away from this sort of functionality https://github.com/junit-team/junit5/pull/1422#issuecomment-389644868

I'm ok with accepting this test. Can you leave a comment in the code about this experimentally for us to refer to when we upgrade?

Comment (nikzaugg) [reply src/test/java/edu/harvard/iq/dataverse/mydata/MyDataUtilTest.java]: hi @matthew-a-dunlap 

Thanks for the review. I added a comment about JUnit Theories in a new commit.

Comment (poikilotherm) [reply src/test/java/edu/harvard/iq/dataverse/mydata/MyDataUtilTest.java]: Could this be replaced with the upstream-supported Dynamic Testing in JUnit5?

Promising blog article on that: https://blog.codefx.org/libraries/junit-5-dynamic-tests/

**Comment by djbrooke** [comment]
Hi @nikzaugg - is this ready for Code Review?

**Comment by nikzaugg** [comment]
Hi @djbrooke 
Yes, it is ready for review. I added the requested changes with my last commit.

**Comment by pdurbin** [comment]
Should a new issue be created since #5634 is already closed?

**Comment by matthew-a-dunlap** [comment]
#5643 is good and ready for QA

**Comment by djbrooke** [comment]
@pdurbin I'm OK with this moving as a PR. Thanks @matthew-a-dunlap for the code review!

**Pull request merged**